### ISSUE TITLE
Saving notes in the digi app does not work correctly

### ISSUE THREAD

**Issue by AstridBVW** [NHMDenmark/Mass-Digitizer#486]
**Describe the bug**
While digitising we are having problems with information added to the notes field in the digi app not being saved even though the "SAVE" button is used. 

**To Reproduce**
1. Log into the digi app and fill out the standard fields (screenshot 1)
2. Scan a barcode
3. The digi app automatically saves and goes on to the next one. Click on "GO BACK"
4. Move the cursor to the Notes field and scan the controlled vocabulary data matrix for "Associated preparation slide #" to add it  to the field. Add the slide number after "#" by typing. Then click "SAVE" (screenshot 2)
5. Go to the database in DB Browser for SQLite and refresh. 
6. Navigate to the notes column in the database. Here you will see that only the information from the data matrix was saved and not the slide number (screenshot 3)
7. Go to the digi app and click on "SAVE" again.
8. Go to the database in DB Browser for SQLite and refresh. You will now see that the slide number is saved as well (screenshot 4)

**Expected behavior**
I expected all of the information in the notes field to be saved by clicking on the "SAVE" button once.

**Screenshots**
<img width="960" alt="Skærmbillede 2024-02-15 112414" src="https://github.com/NHMDenmark/Mass-Digitizer/assets/97025379/ba3458e1-b72b-43a8-b1d3-153d65d2de46">
<img width="960" alt="Skærmbillede 2024-02-15 112641" src="https://github.com/NHMDenmark/Mass-Digitizer/assets/97025379/3f93000b-f98e-4de0-8862-c820a5ec4b31">
<img width="960" alt="Skærmbillede 2024-02-15 112615" src="https://github.com/NHMDenmark/Mass-Digitizer/assets/97025379/8635630c-1c07-429e-ac90-037ae3ce766a">
<img width="960" alt="Skærmbillede 2024-02-15 112717" src="https://github.com/NHMDenmark/Mass-Digitizer/assets/97025379/1b9ab4a1-57a4-4803-96c5-dc29ea9b40ec">

**Please complete the following information:**
 - version 
 - behavior in the database
 - specify relation information 

**Additional context**
The steps above are what happened while I was digitising (AstridBVW). I also tried adding information to the notes field BEFORE scanning the barcode and then the information was saved correctly. @RebekkaML also looked into this while digitising and figured out some more. She found the same as me when adding information to the notes field and then scanning the barcode, but she also found that if you scan the barcode, click "GO BACK", add the information to the notes field, _move the cursor to the barcode field_ and then click"SAVE", it saves correctly. So it seems to be a problem relating to some rule about the barcode field  and saving.

**Comment by FedorSteeman** [comment]
Debugging version 1.2.0 I could not reproduce the phenomenon as it immediately saves the whole value upon the first save:

![image](https://github.com/NHMDenmark/Mass-Digitizer/assets/10909008/c3d0147f-bab7-43a2-a3e5-9d01f0fcccc5)

@AstridBVW  Could you please test again in version 1.2.0  ?

**Comment by PipBrewer** [comment]
Check latest version of App and try and reproduce with and without data matrix

**Comment by AstridBVW** [comment]
I have assigned this to @RebekkaML instead. She is familiar with this ticket and have a version of the app on her computer that she can use for testing (I have a mac).

**Comment by RebekkaML** [comment]
I tested this in version 1.2.0 and could not reproduce this error there either. I tried it both with the "controlled vocabulary data matrix" and by typing the comment manually, and both worked fine. It also didn't make a difference whether I entered the comment first and then the barcode or entered the barcode, saved it and the clicked "GO BACK" to enter the comment, it appeared correctly in the database in all cases.
